### PR TITLE
update typing on HttpResponse and StreamingHttpResponse

### PR DESCRIPTION
**Pull request by terencehonles**
This fixes a regression in https://github.com/typeddjango/django-stubs/pull/695

# Description

While the documentation for `HttpResponse` and `StreamingHttpResponse` *says* `content` and `streaming_content` [should be bytestrings](https://github.com/django/django/blob/ecf87ad513fd8af6e4a6093ed918723a7d88d5ca/django/http/response.py#L324-L327) or [an iterable of bytestrings](https://github.com/django/django/blob/0a28b42b1510b8093a90718bafd7627ed67fa13b/django/http/response.py#L395-L399) respectively, this is not what the API supports ([here](https://github.com/django/django/blob/ecf87ad513fd8af6e4a6093ed918723a7d88d5ca/django/http/response.py#L342-L362) and [here](https://github.com/django/django/blob/0a28b42b1510b8093a90718bafd7627ed67fa13b/django/http/response.py#L415-L427)) and there are tests which make sure the API supports more than bytestrings ([here](https://github.com/django/django/blob/0a28b42b1510b8093a90718bafd7627ed67fa13b/tests/cache/tests.py#L2250), [here](https://github.com/django/django/blob/0a28b42b1510b8093a90718bafd7627ed67fa13b/tests/i18n/urls.py#L8), etc). Before assigning `content` or `streaming_content` the code paths will call  `self.make_bytes` to coerce the value to bytes.

### CODE REVIEW

**Comment by terencehonles** [comment]
It looks like the property change is also captured by https://github.com/typeddjango/django-stubs/pull/393, but the regression is still an issue that this PR only addresses AFAIK

File: django-stubs/http/response.pyi [hunk 1]
```diff
@@ -101,8 +101,11 @@ class HttpResponse(HttpResponseBase):
 
 class StreamingHttpResponse(HttpResponseBase):
     content: Any
-    streaming_content: Iterator[bytes]
-    def __init__(self, streaming_content: Iterable[bytes] = ..., *args: Any, **kwargs: Any) -> None: ...
+    @property
```
Comment (sobolevn) [inline django-stubs/http/response.pyi]: Can you please add a test for this? This seems to be a common issue.

**Review by sobolevn — CHANGES_REQUESTED** [review]
👍

File: django-stubs/http/response.pyi (hunk 1 above)
Comment (terencehonles) [reply django-stubs/http/response.pyi]: yeah I'll get to that

File: django-stubs/http/response.pyi [hunk 2]
```diff
@@ -10,6 +10,31 @@ from django.test.client import Client
 from django.urls import ResolverMatch
 from django.utils.datastructures import CaseInsensitiveMapping
 
+_T = TypeVar('_T')
+_U = TypeVar('_U')
+
+class _PropertyDescriptor(Generic[_T, _U]):
```
Comment (terencehonles) [inline django-stubs/http/response.pyi:16]: I used https://github.com/python/mypy/issues/3004#issuecomment-493741241 as a reason to test to see if I could implement the property with a descriptor. I'm not sure if there are any negative side effects, but properties *are* descriptors and it looks like the types resolve correctly now (my original code using properties did not work when adding the tests)

File: django-stubs/http/response.pyi [hunk 3]
```diff
@@ -21,7 +46,7 @@ class ResponseHeaders(CaseInsensitiveMapping):
     def pop(self, key: str, default: Optional[str] = ...) -> str: ...
     def setdefault(self, key: str, value: str) -> None: ...
 
-class HttpResponseBase(Iterable[Any]):
+class HttpResponseBase:
```
Comment (terencehonles) [inline django-stubs/http/response.pyi:49]: This is not iterable directly, but its subclasses are

**Comment by terencehonles** [comment]
This looks like this PR actually completely superceeds #393 since `getvalue()` is already typed from https://github.com/typeddjango/django-stubs/pull/422

**Comment by terencehonles** [comment]
@sobolevn looks like the CI is now passing again, so a final review would be appreciated (just letting you know I'm done, but not a sense of urgency)

**Review by sobolevn — APPROVED** [review]
Awesome work! Thank you! 👍

**Pull request merged**